database/service_instance.go

```diff
@@ -787,6 +797,60 @@ func (c *ServiceInstanceClient) WaitForServiceInstanceDeleted(input *GetServiceI
 	})
 }
 
+// UpdateServiceInstanceInput defines the attributes available to update for a service instance
+type UpdateServiceInstanceInput struct {
+	// Name of the service instance to update.
+	// Required
+	Name string `json:"-"`
+	// Specify size of additional storage in Giga Bytes. This parameter is optional. User can change shape
+	// only without adding storage. If additionalStorage is specified, minimum value is 1GB and maximum value is 1TB.
+	// Optional
+	AdditionalStorage string `json:"additionalStorage,omitempty"`
+	// (Applies only to service instances that use Oracle RAC and Oracle Data Guard together.)
+	// Specifies whether the scaling operation applies to the primary database or standby database of the
+	// Data Guard configuration. Specify the value DB_1 for the primary database or the value DB_2 for the
+	// standby database.
+	// Optional
+	ComponentInstanceName string `json:"componentInstanceName,omitempty"`
+	// Specify new shape for the Database Cloud Service instance. User can specify a higher shape (Scale Up) or
+	// a lower shape (Scale Down). Shape is optional. User can add storage only without changing shape.
+	// Optional
+	Shape ServiceInstanceShape `json:"shape,omitempty"`
+	// This parameter specifies usage of additional storage and is applicable only when additionalStorage is
+	// specified. Specify usage to extend Data or Backup storage volumes of Database Cloud Service instance.
+	// Valid values are data to extend Data Storage Volume and fra to extend Backup Storage Volume. If usage is
+	// not specified, new storage volume will be created.
+	// Optional
+	Usage ServiceInstanceUsage `json:"usage,omitempty"`
+}
+
+// UpdateServiceInstance updates the specified service instance
+func (c *ServiceInstanceClient) UpdateServiceInstance(input *UpdateServiceInstanceInput) (*ServiceInstance, error) {
+	if c.PollInterval == 0 {
+		c.PollInterval = WaitForServiceInstanceReadyPollInterval
+	}
+	if c.Timeout == 0 {
+		c.Timeout = WaitForServiceInstanceReadyTimeout
+	}
+
+	if err := c.updateResource(input.Name, *input, nil); err != nil {
+		return nil, err
+	}
+
+	// Call wait for instance ready now, as creating the instance is an eventually consistent operation
```

Comment:
I think this should either be "Call WaitForInstanceRunning now" or possibly just reworded.

Reply:
Good call!